rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,224 @@
+            # chkstat prefers the new paths over the old ones, so callers that only care about the
+            # first matching file must mimic that.
+            yield '/usr/share/permissions/' + name
+            yield '/etc/' + name
+
+    def _parsefile(self, fname):
+        lastfn = None
+        with open(fname) as inputfile:
+            for lnr, line in enumerate(inputfile, start=1):
+                line = line.split('#')[0].strip()
+                if not line:
+                    continue
+
+                if line.startswith('+capabilities '):
+                    line = line[len('+capabilities '):]
+                    if lastfn:
+                        self.perms[lastfn]['fscaps'] = line
+                    continue
+
+                line = line.split()
+                if len(line) == 3:
+                    fn = line[0]
+                    owner = line[1].replace('.', ':')
+                    mode = line[2]
+
+                    self.perms[fn] = {'owner': owner, 'mode': stat.S_IMODE(int(mode, 8))}
+                    # for permissions that don't change and therefore
+                    # don't need special handling
+                    if fname in self._paths_to('permissions'):
+                        self.perms[fn]['static'] = True
+
+                    lastfn = fn
+                else:
+                    lastfn = None
+                    print(f'{fname}: Malformatted line {lnr}: {" ".join(line)}...', file=sys.stderr)
+
+    def check_binary(self, pkg):
+        permfiles = set()
+        # first pass, find permissions.d files and check against allow list
+        for f, finfo in pkg.files.items():
+            for prefix in self._paths_to('permissions.d/'):
+                if f.startswith(prefix):
+                    if finfo.is_ghost:
+                        self.output.add_info('E', pkg, 'permissions-ghostfile', f)
+                        continue
+
+                    bn = f[len(prefix):]
+                    if bn not in _permissions_d_allowed:
+                        self.output.add_info('E', pkg, 'permissions-unauthorized-file', f)
+
+                    bn = 'permissions.d/' + bn.split('.')[0]
+                    permfiles.add(bn)
+
+        # parse found permissions.d files
+        for f in permfiles:
+            # check for a .secure file first, falling back to the plain file
+            for path in self._paths_to(f + '.secure', f):
+                if path in pkg.files:
+                    self._parsefile(pkg.dirName() + path)
+                    break
+
+        need_set_permissions = False
+        found_suseconfig = False
+        # second pass, find permissions violations
+        for f, pkgfile in pkg.files.items():
+
+            if pkgfile.filecaps:
+                self.output.add_info('E', pkg, 'permissions-fscaps', f'{f} has fscaps "{pkgfile.filecaps}"')
+
+            mode = pkgfile.mode
+            owner = pkgfile.user + ':' + pkgfile.group
```

Comment:
f'{pkgfile.user}:{pkgfile.group}' ?

Reply:
I would prefer to stay with `pkgfile.user + ':' + pkgfile.group`.